shell/components/cards/AppCard.vue

```diff
@@ -0,0 +1,499 @@
+<script setup>
+import LazyImage from '@shell/components/LazyImage';
+
+const emit = defineEmits(['card-click', 'repo-click', 'category-click', 'tag-click']);
+
+defineProps({
+  title: {
+    type:     String,
+    required: true
+  },
+  description: {
+    type:     String,
+    required: true
+  },
+  logo: {
+    type:    String,
+    default: ''
+  },
+  logoAltText: {
+    type:    String,
+    default: 'logo'
+  },
+  featured: {
+    type:    Boolean,
+    default: false
+  },
+  asLink: {
+    type:    Boolean,
+    default: false
+  },
+  deprecated: {
+    type:    Boolean,
+    default: false
+  },
+  upgradable: {
+    type:    Boolean,
+    default: false
+  },
+  installed: {
+    type:    Boolean,
+    default: false
+  },
+  version: {
+    type:    String,
+    default: ''
+  },
+  repo: {
+    type:    String,
+    default: ''
+  },
+  categories: {
+    type:    Array,
+    default: () => []
+  },
+  tags: {
+    type:    Array,
+    default: () => []
+  }
+});
+
+function handleCardClick(e) {
+  // Prevent card click if the user clicks on an inner actionable element like repo, category, or tag
+  if (e.target.closest('.no-card-click')) {
+    return;
+  }
+
+  emit('card-click');
+}
+
+</script>
+
+<template>
```

Comment:
once nearer to completion double check with @aalves08 about what updates this should have to be accessible (arias, tab order, etc)